src/controllers/hid/hidioreport.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "controllers/controller.h"
+#include "controllers/hid/hiddevice.h"
+#include "util/compatibility/qmutex.h"
+#include "util/duration.h"
+
+class HidIoReport {
+  public:
+    HidIoReport(const unsigned char& reportId);
+
+    /// Latches new report data, which will later send by the IO thread
+    void latchOutputReport(const QByteArray& data,
+            const mixxx::hid::DeviceInfo& deviceInfo,
+            const RuntimeLoggingCategory& logOutput);
+
+    /// Sends the OutputReport to the HID device, when changed data are latched.
+    /// Returns true if a time consuming hid_write operation was executed.
+    bool sendOutputReport(hid_device* pDevice,
```

Comment:
Please add a comment that the caller is responsible for guaranteeing that pDevice (-> please rename consistently as pHidDevide) is accessed only by the current thread during this operation. Probably rename it as `pExclusiveHidDevice` (.h/.cpp) to emphasize this fact.

The design with the scattered mutexes us confusing.

Reply:
Now the hidDeviceMutex is passed to the function too, I think this make the code better readable.